LogfileIterator/LogfileIteratorExample.cpp

```diff
@@ -0,0 +1,181 @@
+ * THE SOFTWARE.
+ */
+
+/**
+ * \example LogfileIteratorExample.cpp
+ *
+ * PolySync Logfile CPP API Iterator Example.
+ *
+ * Shows how to use the Logfile API to iterate over records in a Logfile
+ * outside of the normal PolySync Replay time domain. This means that the
+ * Logfile iterator runs as fast as the CPU will allow.
+ *
+ * @file LogfileIteratorExample.cpp
+ * @brief LogfileIterator Source
+ *
+ */
+
+
+#include "LogfileTestNode.hpp"             // For all CPP Logfile API examples.
+
+using namespace std;
+
+using namespace polysync::datamodel;
+
+
+void LogFileTestNode::prepareLogfileToIterate()
+{
+    sleepMicro ( 1000000 ); // Sleep before mode off so queue doesn't flush.
+
+    _logFile->setModeOff(); // Automatically disables state.
+
+    _logFile->setFilePath( "/tmp/polysync_logfile.plog"  );
+
+    cout << "Logfile Iterator started." << endl;
+
+    _logFileWasIterated = true;
+}
+
+
+void LogFileTestNode::logfileIteratorCallback(
+        const ps_logfile_attributes * const fileAttributes,
+        const ps_msg_type msg_type,
+        const ps_rnr_log_record * const logRecord,
+        void * const userData )
+{
+    cout << "\nLogfile iterator callback\n";
+
+    if( logRecord )
+    {
+        cout<<"Log Record: Index: " << logRecord->index <<" - Size: "
+           << logRecord->size <<" - Previous Size: "
+           << logRecord->prev_size <<" - RnR Timestamp ( \nheader.timestamp):"
+           << logRecord->timestamp;
+    }
+
+    cout << endl;
+}
+
+
+
+void LogFileTestNode::iterateOverLogfiles()
+{
+    _logFile->forEachIterator(
+                _logFile->getFilePath().c_str(),
+                &logfileIteratorCallback,
+                nullptr );
+}
+
+
+void LogFileTestNode::printResults()
+{
+    if( _logFileWasIterated )
+    {
+        cout << "\nIterator complete.\n";
+    }
+
+    cout << "\nReleasing logfile resources. If all messages did not print "
+            "to Terminal \nin either read, write, or iterator, that is due to "
+            "i/o (printf / cout) \nbeing slower than CPU.\n\n";
+}
+
+
+void LogFileTestNode::initStateEvent()
+{
+    // 1. Init LogFile API resources:
+    _logFile = new Logfile{ *this };
+
+    // 2. Set up parameters.
+    prepareLogfileToIterate();
+
+    /* 3. Iterate over Logfiles in initStateEvent(), not okStateEvent(), as
+     * Logfile iterator operates outside of the normal Replay time domain.
+     */
+    iterateOverLogfiles();
+
+    // 4. Disconnect after Iterator is done iterating over a Logfile.
+    disconnectPolySync();
+}
+
+
+void LogFileTestNode::okStateEvent()
+{
+    // When using Logfile Iterator, no code should be in in okStateEvent().
+}
+
+
+void LogFileTestNode::releaseStateEvent()
+{
+    printResults();
+
+    /* Turn off mode. Turning off the mode automatically disables state.
+     * Sleep before turning mode off after last write to avoid flushing of queue.
+     */
+    sleep( 5 );
```

Comment:
what's the reason you're not using psync_sleep_micro? The psync call plays with our setup.